File: das_backend/src/test/java/ch/sbb/backend/admin/domain/repository/DomainServicePointServiceTest.java
package ch.sbb.backend.admin.domain.repository;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import ch.sbb.backend.admin.domain.model.ServicePoint;
import ch.sbb.backend.admin.domain.service.DomainServicePointService;
import java.util.List;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class DomainServicePointServiceTest {

    private ServicePointRepository servicePointRepository;
    private DomainServicePointService underTest;

    @BeforeEach
    void setUp() {
        servicePointRepository = mock(ServicePointRepository.class);
        underTest = new DomainServicePointService(servicePointRepository);
    }

    @Test
    void shouldUpdateServicePoints_thenSave() {
        var servicePoints = List.of(new ServicePoint(1, "designation", "abbreviation"));
        underTest.updateAll(servicePoints);
        verify(servicePointRepository).saveAll(servicePoints);
    }
}

File: backend/src/test/kotlin/ch/sbb/backend/servicepoints/domain/repository/DomainServicePointServiceTest.kt
package ch.sbb.backend.servicepoints.domain.repository

import ch.sbb.backend.servicepoints.domain.ServicePoint
import ch.sbb.backend.servicepoints.domain.service.DomainServicePointService
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify


class DomainServicePointServiceTest {

    private lateinit var servicePointRepository: ServicePointRepository
    private lateinit var tested: DomainServicePointService

    @BeforeEach
    fun setUp() {
        servicePointRepository = mock(ServicePointRepository::class.java)
        tested = DomainServicePointService(servicePointRepository)
    }

    @Test
    fun shouldUpdateServicePoints_thenSave() {
        val servicePoints: List<ServicePoint> =
            listOf(ServicePoint(1, "designation", "abbreviation"))
        tested.updateAll(servicePoints)
        verify(servicePointRepository).saveAll(servicePoints)

    }
}
